src/machines/java/com/enderio/machines/common/io/item/MachineInventory.java

```diff
@@ -32,6 +35,9 @@ public MachineInventory(IIOConfig config, MachineInventoryLayout layout) {
         this.config = config;
         this.layout = layout;
     }
+    public void addSlotChangedCallback(IntConsumer callback) {
```

Comment:
is this really needed? Can't we just use an anon impl of onContentChanged?

Reply:
on content changed would run too often, I'm only interested in if the item in the slot has changed (like air -> planks, planks -> air, air -> stone). on content changed doesn't have enough info to determine if the item has changed.

Reply:
Hmm you mean a potential planks -> planks case could happen?

Reply:
yes, that as well

Reply:
Is this discussion resolved? Thanks :)